### PR TITLE
Feat/formcontrol examples

### PR DESCRIPTION
**Pull request by akash3gtm**
added all the formcontrol examples

### CODE REVIEW

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx [hunk 1]
```diff
@@ -275,6 +301,607 @@ The following are the lists of all the libraries and packages the component reli
...
+    metaData={{
+      code: `
+        <FormControl>
+          <FormControl.Label mb='$1'>
+            <FormControl.Label.Text color='$text500' lineHeight='$md'>
+              Comment
+            </FormControl.Label.Text>
+          </FormControl.Label>
+          <TextArea>
+            <TextArea.Input pt='$2' pr='$1'/>
```
Comment (amars29) [inline example/storybook/src/components/Forms/FormControl/index.stories.mdx:508]: Reason for right padding?

Comment (akash3gtm) [reply example/storybook/src/components/Forms/FormControl/index.stories.mdx:508]: to get the padding exact as the design.
in figma, there is different padding for left and right side inside the textArea.

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx [hunk 2]
```diff
@@ -275,6 +301,607 @@ The following are the lists of all the libraries and packages the component reli
...
+
+#### Form Control with Input
+
+<AppProvider>
+  <CodePreview
+    showComponentRenderer={true}
+    showArgsController={false}
+    metaData={{
+      code: `
+        <FormControl minWidth='$80'>
```
Comment (amars29) [inline example/storybook/src/components/Forms/FormControl/index.stories.mdx:316]: readon for minWidth?

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx [hunk 3]
```diff
@@ -275,6 +301,607 @@ The following are the lists of all the libraries and packages the component reli
...
+          <FormControl.Label mb='$1'>
+            <FormControl.Label.Text color='$text500' lineHeight='$md'>
+              Name
+            </FormControl.Label.Text>
+          </FormControl.Label>
+          <Input>
+            <Input.Input type="text"  py='$2' px='$3' />
```
Comment (amars29) [inline example/storybook/src/components/Forms/FormControl/index.stories.mdx:323]: there is no padding given in the design

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx [hunk 4]
```diff
@@ -275,6 +301,607 @@ The following are the lists of all the libraries and packages the component reli
...
+
+<AppProvider>
+  <CodePreview
+    showComponentRenderer={true}
+    showArgsController={false}
+    metaData={{
+      code: `
+        <FormControl>
+          <FormControl.Label>
+            <Heading color='$text900' fontSize='$md' lineHeight='$md'>
```
Comment (amars29) [inline example/storybook/src/components/Forms/FormControl/index.stories.mdx:357]: Make sure that it is not rendering h1 on web.

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx [hunk 5]
```diff
@@ -275,6 +301,607 @@ The following are the lists of all the libraries and packages the component reli
...
```
Comment (amars29) [inline example/storybook/src/components/Forms/FormControl/index.stories.mdx:430]: Make sure that it is not rendering h1 on web.

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx (hunk 1 above)
Comment (amars29) [reply example/storybook/src/components/Forms/FormControl/index.stories.mdx:508]: design has been corrected.

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx [hunk 6]
```diff
@@ -275,6 +301,607 @@ The following are the lists of all the libraries and packages the component reli
...
+              </Select.Content>
+            </Select.Portal>
+          </Select>
+          <Button px='$3' bg='$darkBlue600'>
+            <Button.Text fontSize='$sm' fontWeight='$medium'>
+              Next
+            </Button.Text>
+          </Button>
+        </HStack>
+        <FormControl.Error mt='$2'>
```
Comment (amars29) [inline example/storybook/src/components/Forms/FormControl/index.stories.mdx:713]: This fold is not present in the design. Maybe got copied from above example code. Please remove that

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx [hunk 7]
```diff
@@ -275,6 +301,607 @@ The following are the lists of all the libraries and packages the component reli
...
+    showArgsController={false}
+    metaData={{
+      code: `
+      function App() {
+        const [showModal, setShowModal] = useState(false);
+        return(
+          <Wrapper>
+            <Center h={500}>
+              <Button
+                py='$2.5'
```
Comment (amars29) [inline example/storybook/src/components/Forms/FormControl/index.stories.mdx:752]: Please match the padding

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx [hunk 8]
```diff
@@ -275,6 +301,607 @@ The following are the lists of all the libraries and packages the component reli
...
+              </Button>
+              <Modal
+                isOpen={showModal}
+                onClose={() => {
+                  setShowModal(false)
+                }}
+              >
+                <Modal.Content maxWidth='$96' bg='$white'>
+                  <Modal.Body p='$5'>
+                    <VStack space='xs' mb='$4'>
```
Comment (amars29) [inline example/storybook/src/components/Forms/FormControl/index.stories.mdx:771]: Reason for mb="$4"

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx [hunk 9]
```diff
@@ -275,6 +301,607 @@ The following are the lists of all the libraries and packages the component reli
...
+                      <VStack py='$2' space='xl'>
+                        <VStack space='xs'>
+                          <FormControl.Label.Text color='$text500'>
+                            Name
+                          </FormControl.Label.Text>
+                          <Input>
+                            <Input.Input
+                              type="text"
+                              py='$2'
+                              px='$3'
```
Comment (amars29) [inline example/storybook/src/components/Forms/FormControl/index.stories.mdx:789]: px="$3" is the by default style. we shouldn't explicitly provided the styled that are already present in the theme.

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx [hunk 10]
```diff
@@ -275,6 +301,607 @@ The following are the lists of all the libraries and packages the component reli
...
+                          </Input>
+                        </VStack>
+                        <VStack space='xs'>
+                          <FormControl.Label>
+                            <FormControl.Label.Text color='$text500'>
+                              Email
+                            </FormControl.Label.Text>
+                          </FormControl.Label>
+                          <Input>
+                            <Input.Input
```
Comment (amars29) [inline example/storybook/src/components/Forms/FormControl/index.stories.mdx:801]: we shouldn't explicitly provided the styled that are already present in the theme.

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx (hunk 1 above)
Comment (akash3gtm) [reply example/storybook/src/components/Forms/FormControl/index.stories.mdx:508]: updated

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx (hunk 2 above)
Comment (akash3gtm) [reply example/storybook/src/components/Forms/FormControl/index.stories.mdx:316]: removed, it was extra.

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx (hunk 4 above)
Comment (akash3gtm) [reply example/storybook/src/components/Forms/FormControl/index.stories.mdx:357]: checked, rendering h4

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx (hunk 6 above)
Comment (akash3gtm) [reply example/storybook/src/components/Forms/FormControl/index.stories.mdx:713]: updated

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx (hunk 8 above)
Comment (akash3gtm) [reply example/storybook/src/components/Forms/FormControl/index.stories.mdx:771]: it should be padding, updated.

File: example/storybook/src/components/Forms/FormControl/index.stories.mdx [hunk 11]
```diff
@@ -275,6 +301,594 @@ The following are the lists of all the libraries and packages the component reli
...
+#### Form Control with Form Actions
+
+<AppProvider>
+  <CodePreview
+    showComponentRenderer={true}
+    showArgsController={false}
+    metaData={{
+      code: `
+        <FormControl flexDirection='row'>
+          <Button bg='$red700' mr='$4' py='$2.5' px='$3'>
```
Comment (amars29) [inline example/storybook/src/components/Forms/FormControl/index.stories.mdx:866]: Check the hover and active state color of this button. It should be of red shade.

Comment (akash3gtm) [reply example/storybook/src/components/Forms/FormControl/index.stories.mdx:866]: checked, the color is the same for hover state.

Comment (amars29) [reply example/storybook/src/components/Forms/FormControl/index.stories.mdx]: change it to darker shade

**Pull request closed without merging**